test/e2e/storage/persistent_volumes-local.go

```diff
@@ -212,89 +220,58 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 		})
 
 		It("should be able to mount volume in two pods one after other, write from pod1, and read from pod2", func() {
-			By("Creating pod1")
-			pod1, pod1Err := createLocalPod(config, testVol)
-			Expect(pod1Err).NotTo(HaveOccurred())
-
-			framework.ExpectNoError(framework.WaitForPodRunningInNamespace(config.client, pod1))
-			pod1NodeName, pod1NodeNameErr := podNodeName(config, pod1)
-			Expect(pod1NodeNameErr).NotTo(HaveOccurred())
-			framework.Logf("Pod1 %q created on Node %q", pod1.Name, pod1NodeName)
-			Expect(pod1NodeName).To(Equal(node0.Name))
-
-			writeCmd, readCmd := createWriteAndReadCmds(volumeDir, testFile, testVol.hostDir /*writeTestFileContent*/)
-
-			By("Writing in pod1")
-			podRWCmdExec(pod1, writeCmd)
+			twoPodsReadWriteSerialTest(config, testVol)
+		})
+	})
 
-			By("Deleting pod1")
-			framework.DeletePodOrFail(config.client, config.ns, pod1.Name)
+	Context("when two pods request one prebound PVC at the same time", func() {
 
-			By("Creating pod2")
-			pod2, pod2Err := createLocalPod(config, testVol)
-			Expect(pod2Err).NotTo(HaveOccurred())
+		var testVol *localTestVolume
 
-			framework.ExpectNoError(framework.WaitForPodRunningInNamespace(config.client, pod2))
-			pod2NodeName, pod2NodeNameErr := podNodeName(config, pod2)
-			Expect(pod2NodeNameErr).NotTo(HaveOccurred())
-			framework.Logf("Pod2 %q created on Node %q", pod2.Name, pod2NodeName)
-			Expect(pod2NodeName).To(Equal(node0.Name))
+		BeforeEach(func() {
+			testVol = setupLocalVolumePVCPV(config, DefaultMountPointType)
+		})
 
-			By("Reading in pod2")
-			readOut := podRWCmdExec(pod2, readCmd)
-			Expect(readOut).To(ContainSubstring(testVol.hostDir)) /*aka writeTestFileContents*/
+		AfterEach(func() {
+			cleanupLocalVolume(config, testVol)
+		})
 
-			By("Deleting pod2")
-			framework.DeletePodOrFail(config.client, config.ns, pod2.Name)
+		It("should be able to mount volume in two pods at the same time, write from pod1, and read from pod2", func() {
+			twoPodsReadWriteTest(config, testVol)
 		})
 	})
 
-	Context("when two pods request one prebound PVC at the same time", func() {
+	Context("with true tmpfs mountpoint when two pods request one prebound PVC one after other", func() {
```

Comment:
@msau42 - I'm re-writing the code with some nested loops to allow easy addition of more tests and more mountpoint types.